### ISSUE TITLE
🐛 `biome_grit_parser::parse_grit` panics

### ISSUE THREAD

**Issue by Arian8j2** [biomejs/biome#4825]
### Environment information

```block
CLI:
  Version:                      0.0.0
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           linux

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           unset
  JS_RUNTIME_NAME:              unset
  NODE_PACKAGE_MANAGER:         unset

Biome Configuration:
  Status:                       unset

Workspace:
  Open Documents:               0

I compiled the `biome` from `6ea885f0cec47a6db48d662cbd828f4f335209ba` commit
```

### What happened?

1. create a file named `suspect1.grit` and put this `grit` query in it:
```grit
engine marzano(0.1)
language python

`$var in $dict.keys()` => `$var in $dict`
```
2. then running `biome format` will result in:
```
Biome encountered an unexpected error

This is a bug in Biome, not an error in your code, and we would appreciate it if you could report it to https://github.com/biomejs/biome/issues/ along with the following information to help us fixing the issue:

Source Location: /path_to_biome/crates/biome_parser/src/diagnostic.rs:350:14
Thread Name: biome::worker_23
Message: Expected token to be a punctuation or keyword.

/path_to_current_folder/suspect1.grit internalError/panic  INTERNAL  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ processing panicked: Expected token to be a punctuation or keyword.

  ⚠ This diagnostic was derived from an internal Biome error. Potential bug, please report it if necessary.


Checked 0 files in 1010µs. No fixes applied.
internalError/io ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ No files were processed in the specified paths.

```
i traced the panic from calling `biome_grit_parser::parse_grit` and the result is this:
```
stack backtrace:
   4: core::option::Option<T>::expect
             at /.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/option.rs:933:21
   5: biome_parser::diagnostic::expected_token
             at /.cargo/git/checkouts/biome-d49ce8898b420a50/6ea885f/crates/biome_parser/src/diagnostic.rs:348:9
   6: biome_parser::Parser::expect
             at /.cargo/git/checkouts/biome-d49ce8898b420a50/6ea885f/crates/biome_parser/src/lib.rs:422:24
   7: biome_grit_parser::parser::parse_root
             at /.cargo/git/checkouts/biome-d49ce8898b420a50/6ea885f/crates/biome_grit_parser/src/parser/mod.rs:91:5
   8: biome_grit_parser::parse_grit_with_cache::{{closure}}
             at /.cargo/git/checkouts/biome-d49ce8898b420a50/6ea885f/crates/biome_grit_parser/src/lib.rs:26:9
   9: tracing::span::Span::in_scope
             at /.cargo/registry/src/index.crates.io-6f17d22bba15001f/tracing-0.1.41/src/span.rs:1102:9
  10: biome_grit_parser::parse_grit_with_cache
             at /.cargo/git/checkouts/biome-d49ce8898b420a50/6ea885f/crates/biome_grit_parser/src/lib.rs:23:5
  11: biome_grit_parser::parse_grit
             at /.cargo/git/checkouts/biome-d49ce8898b420a50/6ea885f/crates/biome_grit_parser/src/lib.rs:18:5
```
it's not just this pattern that makes it panic, there are lots of other patterns that trigger it too, you can test patterns in [stdlib](https://github.com/getgrit/stdlib)

### Expected result

It shouldn't panic

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by ematipico** [comment]
Here's a [reproduction](https://biomejs.dev/playground/?lineWidth=120&quoteStyle=single&indentWidth=4&arrowParentheses=as-needed&files.main.grit=ZQBuAGcAaQBuAGUAIABtAGEAcgB6AGEAbgBvACgAMAAuADEAKQAKAGwAYQBuAGcAdQBhAGcAZQAgAHAAeQB0AGgAbwBuAAoACgBgACQAdgBhAHIAIABpAG4AIAAkAGQAaQBjAHQALgBrAGUAeQBzACgAKQBgACAAPQA%2BACAAYAAkAHYAYQByACAAaQBuACAAJABkAGkAYwB0AGAA).

@Arian8j2, please next time, provide one :)

**Comment by arendjr** [comment]
I think this happens because I made a bit of a shortcut with the parser, which only supports `engine biome` and the languages we support in Biome. But I agree it’s an unfortunate choice if someone wants to use our formatter with Grit files not intended for Biome plugins, which is a totally valid use case.

Fortunately it shouldn’t be too hard to allow arbitrary engine and languages names, but it requires a small extension to the grammar.

**Comment by morgante** [comment]
I'm working on this, should have a PR shortly.
